### PR TITLE
Xcode error message

### LINKED ISSUE #22536: `flutter run` doesn't show some critical log items from Xcode
## Steps to Reproduce

1. Create an app using Firestore (`cloud_firestore` in dependencies). For example, the official [Firebase for Flutter codelab](https://codelabs.developers.google.com/codelabs/flutter-firebase/index.html?index=..%2F..%2Findex#0).
2. Neglect to set up the iOS project with `GoogleService-Info.plist`.
3. `flutter run` the app on iOS.

## Logs

This is what `flutter run` shows, in its entirety:

```
Launching lib/main.dart on iPhone X in debug mode...
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010930d29b __exceptionPreprocess + 331
	1   libobjc.A.dylib                     0x00000001088a9735 objc_exception_throw + 48
	2   CoreFoundation                      0x000000010930d0f5 +[NSException raise:format:] + 197
	3   Runner                              0x0000000104947aa1 +[FIRApp configure] + 481
	4   Runner                              0x0000000104c9d22c -[FLTCloudFirestorePlugin init] + 188
	5   Runner                              0x0000000104c9d0d9 +[FLTCloudFirestorePlugin registerWithRegistrar:] + 297
	6   Runner                              0x00000001048f06bb +[GeneratedPluginRegistrant registerWithRegistry:] + 123
	7   Runner                        <…>
Starting Xcode build...
Xcode build done.
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010665829b __exceptionPreprocess + 331
	1   libobjc.A.dylib                     0x0000000105bf4735 objc_exception_throw + 48
	2   CoreFoundation                      0x00000001066580f5 +[NSException raise:format:] + 197
	3   Runner                              0x0000000101befaa1 +[FIRApp configure] + 481
	4   Runner                              0x0000000101f4522c -[FLTCloudFirestorePlugin init] + 188
	5   Runner                              0x0000000101f450d9 +[FLTCloudFirestorePlugin registerWithRegistrar:] + 297
	6   Runner                              0x0000000101b986bb +[GeneratedPluginRegistrant registerWithRegistry:] + 123
	7   Runner                        <…>
```

Here's what you get in Xcode:

```
2018-10-01 16:30:54.093692-0700 Runner[21990:12461249]  - <AppMeasurement>[I-ACS036002] Analytics screen reporting is enabled. Call +[FIRAnalytics setScreenName:setScreenClass:] to set the screen name or override the default screen class name. To disable screen reporting, set the flag FirebaseScreenReportingEnabled to NO (boolean) in the Info.plist
2018-10-01 16:30:54.370913-0700 Runner[21990:12460659] You've implemented -[<UIApplicationDelegate> application:performFetchWithCompletionHandler:], but you still need to add "fetch" to the list of your supported UIBackgroundModes in your Info.plist.
2018-10-01 16:30:54.371112-0700 Runner[21990:12460659] You've implemented -[<UIApplicationDelegate> application:didReceiveRemoteNotification:fetchCompletionHandler:], but you still need to add "remote-notification" to the list of your supported UIBackgroundModes in your Info.plist.
2018-10-01 16:30:54.385564-0700 Runner[21990:12460659] *** Terminating app due to uncaught exception 'com.firebase.core', reason: '`[FIRApp configure];` (`FirebaseApp.configure()` in Swift) could not find a valid GoogleService-Info.plist in your project. Please download one from https://console.firebase.google.com/.'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010ac0529b __exceptionPreprocess + 331
	1   libobjc.A.dylib                     0x000000010a1a1735 objc_exception_throw + 48
	2   CoreFoundation                      0x000000010ac050f5 +[NSException raise:format:] + 197
	3   Runner                              0x0000000105fe9aa1 +[FIRApp configure] + 481
	4   Runner                              0x000000010633f22c -[FLTCloudFirestorePlugin init] + 188
	5   Runner                              0x000000010633f0d9 +[FLTCloudFirestorePlugin registerWithRegistrar:] + 297
	6   Runner                              0x0000000105f926bb +[GeneratedPluginRegistrant registerWithRegistry:] + 123
	7   Runner                              0x0000000105f927ca $S6Runner11AppDelegateC11application_29didFinishLaunchingWithOptionsSbSo13UIApplicationC_SDySo0j6LaunchI3KeyaypGSgtF + 138
	8   Runner                              0x0000000105f92bb4 $S6Runner11AppDelegateC11application_29didFinishLaunchingWithOptionsSbSo13UIApplicationC_SDySo0j6LaunchI3KeyaypGSgtFTo + 228
	9   UIKitCore                           0x00000001133ac4d7 -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 280
	10  UIKitCore                           0x00000001133adec4 -[UIApplication _callInitializationDelegatesForMainScene:transitionContext:] + 3979
	11  UIKitCore                           0x00000001133b3527 -[UIApplication _runWithMainScene:transitionContext:completion:] + 1623
	12  UIKitCore                           0x0000000113c55e3e __111-[__UICanvasLifecycleMonitor_Compatability _scheduleFirstCommitForScene:transition:firstActivation:completion:]_block_invoke + 866
	13  UIKitCore                           0x0000000113c52403 +[_UICanvas _enqueuePostSettingUpdateTransactionBlock:] + 153
	14  UIKitCore                           0x0000000113c55a7b -[__UICanvasLifecycleMonitor_Compatability _scheduleFirstCommitForScene:transition:firstActivation:completion:] + 233
	15  UIKitCore                           0x0000000113c56435 -[__UICanvasLifecycleMonitor_Compatability activateEventsOnly:withContext:completion:] + 1085
	16  UIKitCore                           0x0000000113c13b31 __82-[_UIApplicationCanvas _transitionLifecycleStateWithTransitionContext:completion:]_block_invoke + 795
	17  UIKitCore                           0x0000000113c137db -[_UIApplicationCanvas _transitionLifecycleStateWithTransitionContext:completion:] + 435
	18  UIKitCore                           0x0000000113c5892c __125-[_UICanvasLifecycleSettingsDiffAction performActionsForCanvas:withUpdatedScene:settingsDiff:fromSettings:transitionContext:]_block_invoke + 584
	19  UIKitCore                           0x0000000113c59256 _performActionsWithDelayForTransitionContext + 100
	20  UIKitCore                           0x0000000113c5868b -[_UICanvasLifecycleSettingsDiffAction performActionsForCanvas:withUpdatedScene:settingsDiff:fromSettings:transitionContext:] + 221
	21  UIKitCore                           0x0000000113c51aa1 -[_UICanvas scene:didUpdateWithDiff:transitionContext:completion:] + 392
	22  UIKitCore                           0x00000001133b1d46 -[UIApplication workspace:didCreateScene:withTransitionContext:completion:] + 515
	23  UIKitCore                           0x00000001133e3856 -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:] + 357
	24  FrontBoardServices                  0x00000001166aedda -[FBSSceneImpl _didCreateWithTransitionContext:completion:] + 448
	25  FrontBoardServices                  0x00000001166b9f43 __56-[FBSWorkspace client:handleCreateScene:withCompletion:]_block_invoke_2 + 271
	26  FrontBoardServices                  0x00000001166b963a __40-[FBSWorkspace _performDelegateCallOut:]_block_invoke + 53
	27  libdispatch.dylib                   0x000000010c64e63e _dispatch_client_callout + 8
	28  libdispatch.dylib                   0x000000010c651bb4 _dispatch_block_invoke_direct + 301
	29  FrontBoardServices                  0x00000001166ee3c8 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 30
	30  FrontBoardServices                  0x00000001166ee080 -[FBSSerialQueue _performNext] + 457
	31  FrontBoardServices                  0x00000001166ee660 -[FBSSerialQueue _performNextFromRunLoopSource] + 45
	32  CoreFoundation                      0x000000010ab68b31 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	33  CoreFoundation                      0x000000010ab683a3 __CFRunLoopDoSources0 + 243
	34  CoreFoundation                      0x000000010ab62a4f __CFRunLoopRun + 1263
	35  CoreFoundation                      0x000000010ab62221 CFRunLoopRunSpecific + 625
	36  GraphicsServices                    0x000000010fcd01dd GSEventRunModal + 62
	37  UIKitCore                           0x00000001133b5115 UIApplicationMain + 140
	38  Runner                              0x0000000105f92d84 main + 68
	39  libdyld.dylib                       0x000000010c6c4551 start + 1
)
2018-10-01 16:30:54.489749-0700 Runner[21990:12461272] flutter: Observatory listening on http://127.0.0.1:49397/
libc++abi.dylib: terminating with uncaught exception of type NSException
2018-10-01 16:30:54.404948-0700 Runner[21990:12461249] 5.6.0 - [Firebase/Core][I-COR000003] The default Firebase app has not yet been configured. Add `[FIRApp configure];` (`FirebaseApp.configure()` in Swift) to your application initialization. Read more: https://goo.gl/ctyzm8.
```

Note the helpful error that is only visible in Xcode:

```
could not find a valid GoogleService-Info.plist in your project. Please download one from https://console.firebase.google.com/
```

### Flutter doctor

```
$ flutter doctor -v
[✓] Flutter (Channel beta, v0.8.2, on Mac OS X 10.13.6 17G65, locale en-US)
    • Flutter version 0.8.2 at /Users/filiph/dev/flutter
    • Framework revision 5ab9e70727 (3 weeks ago), 2018-09-07 12:33:05 -0700
    • Engine revision 58a1894a1c
    • Dart version 2.1.0-dev.3.1.flutter-760a9690c2

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/filiph/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.0, Build version 10A255
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 28.0.2
    • Dart plugin version 181.5616
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[!] IntelliJ IDEA Community Edition (version 2018.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[!] VS Code (version 1.22.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected devices (3 available)
    • Pixel 2 XL • 711KPZK0592485                       • android-arm64 • Android 9 (API 28)
    • iPhone X   • A4DBE660-4C39-4AEA-B258-7755F584BA6E • ios           • iOS 12.0 (simulator)
    • iPhone XR  • 431DF6F4-BF91-4E1D-9AF1-8324CF683D37 • ios           • iOS 12.0 (simulator)

! Doctor found issues in 2 categories.

```

### PR DESCRIPTION
**Pull request by cyanglaz**
Parse issues in xcresult and display them in console.

Fixes https://github.com/flutter/flutter/issues/22536

Sample output:
```
Semantic Issue (Xcode): Use of undeclared identifier 'ss'
/path/ios/Runner/AppDelegate.m:11:56
```

Also in this PR:
1. Updated XCResultGenerator uses a simpler "issuesMap" when parsing issues.
2. Make simulator build to diagnose failure when build fails.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by jmagman** [comment]
You have test failures.

**Comment by cyanglaz** [comment]
@jmagman Tests passed, PTAL

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 1]
```diff
@@ -383,6 +387,36 @@ Future<XcodeBuildResult> buildXcodeProject({
...
+              globals.printWarning('Xcode build issue (${issue.subType}): ${issue.message}');
+              break;
+          }
+        }
+      } else {
+        globals.printTrace('XCResult parsing error: ${xcResult.parsingErrorMessage}');
+      }
+    }
+    // Free pipe file and xcresult bundle.
+    tempDir.deleteSync(recursive: true);
```
[the remark this answers is not in the document]
Comment (jmagman) [reply packages/flutter_tools/lib/src/ios/mac.dart:417]: I mean: you could delete the temp directory in a `try`/`finally` block so if it crashes the directory is removed.
https://github.com/flutter/flutter/blob/840e109e0718361f70bcb23a3980e407de10859b/packages/flutter_tools/bin/fuchsia_tester.dart#L174-L176

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 2]
```diff
@@ -383,6 +387,36 @@ Future<XcodeBuildResult> buildXcodeProject({
...
+          globals.printStatus('Xcode build has no issues!');
+        }
+        for (final XCResultIssue issue in xcResult.issues) {
+          switch (issue.type) {
+            case XCResultIssueType.error:
+              globals.printError('Xcode build issue (${issue.subType}): ${issue.message}');
+              break;
+            case XCResultIssueType.warning:
+              globals.printWarning('Xcode build issue (${issue.subType}): ${issue.message}');
+              break;
```
[the remark this answers is not in the document]
Comment (cyanglaz) [reply packages/flutter_tools/lib/src/ios/mac.dart:409]: done

[the remark this answers is not in the document]
Comment (jmagman) [reply packages/flutter_tools/lib/src/ios/mac.dart]: Can you add examples for the full `flutter build ios` and `flutter build ios --verbose` output of what this kind of error will look like?  I can't picture how it will look in the context of all the Xcode logging and build failure text.

I would know more when I see the full output, but this seems too wordy: 
```
Xcode build issue (Swift Compiler Error): Expected dictionary value type
```

How about:
```
Swift Compiler Error (Xcode): Expected dictionary value type
```

1. Is this message going to be helpful without the class name and line number?  Unfortunately this means parsing the `url` result anchor (which looks like a query, oddly, thanks Xcode...) `file:///path/to/ios/Runner/AppDelegate.swift#CharacterRangeLen=0&EndingColumnNumber=8&EndingLineNumber=11&StartingColumnNumber=8&StartingLineNumber=11`
2. There are likely quite a few error messages we should not display, like `Command PhaseScriptExecution failed with a nonzero exit code` (would happen every time there's a dart compilation issue).  Same for the warning that happens when a plugin supports a lower iOS version than the app.  Let's get the obvious ones, and then we can add more in future PRs as we discover them.

[the remark this answers is not in the document]
Comment (cyanglaz) [reply packages/flutter_tools/lib/src/ios/mac.dart]: 1. 
I think the compiler error usually display at the end of the build, so it looks like:
```
Running Xcode build...                                                  
 └─Compiling, linking and signing...                        759ms
Xcode build done.                                            5.5s
Xcode build issue (Swift Compiler Error): Expected dictionary value type
Xcode build issue (Swift Compiler Error): Method does not override any method from its superclass
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/38/ypkvsh7n1d94tv_njrw0nwrh00l9tg/T/flutter_tools.CtyCRE/flutter_ios_build_temp_dire1EhUm/temporary_xcresult_bundle

    /Users/ychris/tmp/xcresult/xcresult_parsing/ios/Runner/AppDelegate.swift:8:83: error: expected dictionary value type
        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: nil]?
                                                                                      ^
    /Users/ychris/tmp/xcresult/xcresult_parsing/ios/Runner/AppDelegate.swift:6:17: error: method does not override any method from its superclass
      override func application(
      ~~~~~~~~      ^
    remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in parallel

    Result bundle written to path:
    	/var/folders/38/ypkvsh7n1d94tv_njrw0nwrh00l9tg/T/flutter_tools.CtyCRE/flutter_ios_build_temp_dire1EhUm/temporary_xcresult_bundle


════════════════════════════════════════════════════════════════════════════════
Building a deployable iOS app requires a selected Development Team with a 
Provisioning Profile. Please ensure that a Development Team is selected by:
  1- Open the Flutter project's Xcode target with
       open ios/Runner.xcworkspace
  2- Select the 'Runner' project in the navigator then the 'Runner' target
     in the project settings
  3- Make sure a 'Development Team' is selected under Signing & Capabilities > Team. 
     You may need to:
         - Log in with your Apple ID in Xcode first
         - Ensure you have a valid unique Bundle ID
         - Register your device with your Apple Developer Account
         - Let Xcode automatically provision a profile for your app
  4- Build or run your project again

For more information, please visit:
  https://flutter.dev/docs/get-started/install/macos#deploy-to-ios-devices

Or run on an iOS simulator without code signing
════════════════════════════════════════════════════════════════════════════════
Encountered error while building for device.
``` In this instance, the xcresult message is not particularly useful as the more useful messages are displayed at the end.

What do you think about this when parsing the url result:
```
File: file:///path/to/ios/Runner/AppDelegate.swift
StartLineNumber: 11
EndLineNumber: 11
StartColumnNumber: 8
EndColumnNumber: 8
```

I think this detailed compiler error code position is messy enough to deserve a separate PR.

[the remark this answers is not in the document]
Comment (cyanglaz) [reply packages/flutter_tools/lib/src/ios/mac.dart]: 2
Good idea, I can create a separate PR to filter errors we don't want to display to user in non-verbose mode.

**Comment by cyanglaz** [comment]
Created issues to track the 2 sub problems: https://github.com/flutter/flutter/issues/95035, https://github.com/flutter/flutter/issues/95036

File: packages/flutter_tools/lib/src/ios/mac.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmagman) [reply packages/flutter_tools/lib/src/ios/mac.dart]: > In this instance, the xcresult message is not particularly useful as the more useful messages are displayed at the end.


Instead of printing as you have, what if instead add the XCResult stuff to the `XcodeBuildResult`?  And then the caller can choose how and in what order to display this.

https://github.com/flutter/flutter/blob/52ae102f182afaa0524d0d01d21b2d86d15a11dc/packages/flutter_tools/lib/src/commands/build_ios.dart#L266

> File: file:///path/to/ios/Runner/AppDelegate.swift
StartLineNumber: 11
EndLineNumber: 11
StartColumnNumber: 8
EndColumnNumber: 8

Let's copy what Xcode does when it displays, with just the line and column start.
`/path/to/ios/Runner/AppDelegate.swift:11:8`

> https://github.com/flutter/flutter/issues/95035

I don't think we should land this PR without making sure confusing `PhaseScriptExecution` errors shows up with dart compilation errors.  We can polish in subsequent PRs, but a dart issue is a very core and much more common case than a Swift compilation error, for example.

[the remark this answers is not in the document]
Comment (jmagman) [reply packages/flutter_tools/lib/src/ios/mac.dart]: See `diagnoseXcodeBuildFailure` for where it would probably make sense to handle these from a `XcodeBuildResult`.

[the remark this answers is not in the document]
Comment (jmagman) [reply packages/flutter_tools/lib/src/ios/mac.dart]: Also there are a few things in `diagnoseXcodeBuildFailure` that may actually be more accurate once we can parse these xcresults. For example, it just guesses there's a codesigning error sometimes, even when something else really failed.

[the remark this answers is not in the document]
Comment (cyanglaz) [reply packages/flutter_tools/lib/src/ios/mac.dart]: Parsing "url" in this PR: https://github.com/flutter/flutter/pull/95054

File: packages/flutter_tools/lib/src/commands/build_ios.dart [hunk 3]
```diff
@@ -331,6 +332,29 @@ abstract class _BuildIOSSubCommand extends BuildSubCommand {
       );
     }
 
+    final XCResult xcResult = result.xcResult;
+
+    if (xcResult != null) {
+      if (xcResult.parseSuccess) {
+        if (xcResult.issues.isEmpty) {
+          globals.printStatus('Xcode build has no issues!');
+        }
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/commands/build_ios.dart:341]: Let's remove this trace, we should already be able to see similar output from Xcode in verbose mode.

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 4]
```diff
@@ -725,5 +717,34 @@ bool upgradePbxProjWithFlutterAssets(IosProject project, Logger logger) {
   return true;
 }
 
+void _handleXCResultIssue({required XCResultIssue issue, required Logger logger}) {
+  // Issue summary from xcresult.
+  String issueSummary = '';
+  issueSummary += issue.subType ?? 'Unknown';
+  issueSummary += ' (Xcode): ';
+  issueSummary += issue.message ?? '';
+  issueSummary += issue.location != null? '\n$issue.location' : '';
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/mac.dart:726]:
```suggestion
  issueSummary += issue.location != null ? '\n$issue.location' : '';
```

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 5]
```diff
@@ -623,16 +608,18 @@ Future<void> diagnoseXcodeBuildFailure(XcodeBuildResult result, Usage flutterUsa
...
+  final XCResult? xcResult = result.xcResult;
+  if (xcResult == null) {
     return;
   }
+  if (!xcResult.parseSuccess) {
+    globals.printTrace('XCResult parsing error: ${xcResult.parsingErrorMessage}');
+    return;
+  }
+
+  xcResult.issues.forEach( (final XCResultIssue issue) => _handleXCResultIssue(issue: issue, logger: logger));
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/mac.dart:622]: Analyzer didn't like this: 
```
info • Avoid using `forEach` with a function literal • packages/flutter_tools/lib/src/ios/mac.dart:622:19 • avoid_function_literals_in_foreach_calls
```
```suggestion
  for (final XCResultIssue issue in xcResult.issues) {
    _handleXCResultIssue(issue: issue, logger: logger);
  }
```

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 6]
```diff
@@ -686,3 +716,35 @@ bool upgradePbxProjWithFlutterAssets(IosProject project, Logger logger) {
...
+    case XCResultIssueType.error:
+      logger.printError(issueSummary);
+      break;
+    case XCResultIssueType.warning:
+      logger.printWarning(issueSummary);
+      break;
+  }
+
+  // Add more custom output for flutter users.
+  if (issue.message != null && issue.message!.contains('Provisioning profile')) {
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/mac.dart:738]: I think this may also be `No matching provisioning profiles found`, though I didn't remove the profiles on my machine to test. Can you make this case insensitive to make it a little more robust to string changes?
(I lamented to you on chat already that I wish we have error codes to go off of...)

File: packages/flutter_tools/test/commands.shard/hermetic/build_ios_test.dart [hunk 7]
```diff
@@ -91,6 +92,21 @@ void main() {
...
+        'xcrun',
+        'xcresulttool',
+        'get',
+        '--path',
+        _xcBundleFilePath,
+        '--format',
+        'json'
+      ],
+      stdout: stdout,
+      onRun: onRun);
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/hermetic/build_ios_test.dart:107]:
```suggestion
      onRun: onRun,
    );
```

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 8]
```diff
@@ -686,3 +716,35 @@ bool upgradePbxProjWithFlutterAssets(IosProject project, Logger logger) {
   xcodeProjectFile.writeAsStringSync(buffer.toString());
   return true;
 }
+
+void _handleXCResultIssue({required XCResultIssue issue, required Logger logger}) {
+  // Issue summary from xcresult.
+  String issueSummary = '';
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/mac.dart:722]: Instead of `+=` string concatenation, use a `StringBuffer`.

File: packages/flutter_tools/test/commands.shard/hermetic/build_ios_test.dart [hunk 9]
```diff
@@ -91,6 +92,21 @@ void main() {
...
+  FakeCommand _setUpXCResultCommand({String stdout = '', void Function() onRun}) {
+     return FakeCommand(
+      command: const <String>[
+        'xcrun',
+        'xcresulttool',
+        'get',
+        '--path',
+        _xcBundleFilePath,
+        '--format',
+        'json'
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/hermetic/build_ios_test.dart:104]:
```suggestion
        'json',
```

File: packages/flutter_tools/test/commands.shard/hermetic/build_ios_test.dart [hunk 10]
```diff
@@ -281,4 +299,103 @@ void main() {
...
+    expect(testLogger.traceText, contains('xcresult parser: Unrecognized top level json format.'));
+  }, overrides: <Type, Generator>{
+    FileSystem: () => fileSystem,
+    ProcessManager: () => FakeProcessManager.list(<FakeCommand>[
+      xattrCommand,
+      _setUpFakeXcodeBuildHandler(onRun: () {
+        fileSystem.systemTempDirectory.childDirectory(_xcBundleFilePath).createSync();
+      }),
+      _setUpXCResultCommand(onRun: () {
+      }),
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/hermetic/build_ios_test.dart:321]: `onRun` can be null.
```suggestion
      _setUpXCResultCommand(),
```

**Comment by cyanglaz** [comment]
@jmagman Ready to be reviewed again.

File: packages/flutter_tools/test/commands.shard/hermetic/build_ipa_test.dart [hunk 11]
```diff
@@ -337,12 +355,141 @@ void main() {
...
+      _setUpFakeXcodeBuildHandler(exitCode: 1),
+      _setUpXCResultCommand(stdout: kSampleResultJsonWithIssues),
+    ]),
+    Platform: () => macosPlatform,
+    XcodeProjectInterpreter: () => FakeXcodeProjectInterpreterWithBuildSettings(),
+  });
+
+
+  testUsingContext('Extra error message for provision profile issue in xcresulb bundle.', () async {
+      final BuildCommand command = BuildCommand();
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/hermetic/build_ipa_test.dart:467]: Extra indentation.

**Review by jmagman — APPROVED** [review]
LGTM with minor whitespace nit.
